medcat/utils/ner/deid.py

```diff
@@ -62,18 +62,73 @@ def train(self, json_path: Union[str, list, None],
               *args, **kwargs) -> Tuple[Any, Any, Any]:
         return super().train(json_path, *args, train_nr=0, **kwargs)  # type: ignore
 
-    def deid_text(self, text: str, redact: bool = False) -> str:
+    def get_chunks(self, text:str, de_id_pipe,maximum_tkns) -> List[List[Any]]:
+
+        blocks = [[]]
+        if 'Roberta' in de_id_pipe.ner_pipe.tokenizer.__class__.__name__:
+
+            whitespace_pattern = re.compile(r'\s')
+            tok_output = de_id_pipe.ner_pipe.tokenizer(text, return_offsets_mapping=True)
+            for token, (start, end) in zip(tok_output['input_ids'], tok_output['offset_mapping']):
+                if token in [0, 2]:
+                    continue
+                if len(blocks[-1]) == maximum_tkns:
+                    to_append_block = []
+                    idx_chunk = -1
+                    for i in range(len(blocks[-1]) - 1, len(blocks[-1]) - 21, -1):
+                        # This method to avoid sub-word division is specific for RoBERTA's tokenizer (BPE)
+                        if re.search(whitespace_pattern, de_id_pipe.ner_pipe.tokenizer.decode(blocks[-1][i][0],clean_up_tokenization_spaces=False)):
+                            idx_chunk = i
+                            break
+
+                    if idx_chunk != -1:
+                        to_append_block.extend(blocks[-1][i:])
+                        del blocks[-1][idx_chunk:]
+
+                    to_append_block.append((token, (start, end)))
+                    blocks.append(to_append_block)
+
+                else:
+                    blocks[-1].append((token, (start, end)))
+
+        else:
+            # print("***********WARNING:************\nChunking functionality is implemented for RoBERTa model. The model detected is not RoBERTa, chunking omitted.\nPII information MAY BE REVEALED.")
+            warnings.warn("\n\nChunking functionality is implemented for RoBERTa models. The detected model is not RoBERTa, so chunking is omitted. Be cautious, as PII information MAY BE REVEALED.")
```

Comment:
Let us be consistent - create and use a module level `Logger` instead. Take a look at some of the other core modules to see how we generally do it.